ts/features/lollipop/index.ts

```diff
@@ -0,0 +1,57 @@
+import { PublicKey } from "@pagopa/io-react-native-crypto";
+import URLParse from "url-parse";
+import { SignatureAlgorithm } from "../../utils/httpSignature/types/SignatureAlgorithms";
+import { SignatureComponents } from "../../utils/httpSignature/types/SignatureComponents";
+
+export type LollipopConfig = {
+  nonce: string;
+  customSignatures?: ReadonlyArray<string>;
+};
+
+/**
+ * Utility function to forge the `SignatureComponents` based on the provided inputs.
+ */
+export function forgeSignatureComponents(
+  method: string,
+  inputUrl: URLParse,
+  originalUrl: string
+): SignatureComponents {
+  return {
+    method,
+    authority: inputUrl.hostname,
+    path: inputUrl.pathname,
+    requestTarget: originalUrl,
+    scheme: inputUrl.protocol,
+    targetUri: `${inputUrl.protocol}://${inputUrl.hostname}/${originalUrl}`
+  };
+}
+
+/**
+ * Returns the http-signature algorithm used to sign the signature base specified by
+ * the signature-input header.
+ */
+export function getSignAlgorithm(publicKey: PublicKey): SignatureAlgorithm {
+  return publicKey.kty === "EC" ? "ecdsa-p256-sha256" : "rsa-pss-sha256";
+}
+
+/**
+ * Result type of promises returnd by `chainSignPromises`
+ */
+export type SignPromiseResult = {
+  headerIndex: number;
+  headerName: string;
+  headerValue: string;
+  signature: string;
+  "signature-input": string;
+};
+
+/**
+ * Chains all custom sign promises passed as its input array.
+ */
+export function chainSignPromises(
```

Comment:
I suggest to use `TaskEithers` instead of promises, in this way you can chain them using the methods provided by fp-ts lib (`sequence`)

Reply:
The implementation looks indeed nice! Beware that you're calling the Promises in parallel swallowing all eventual errors (falling back to the empty array, I don't know if it's what the UX is expecting). 

My concern was about the fact that favoring `TaskEithers` over `Promises` is a general practice that is often applied when developing the app backend (but I'm not familiar with the mobile app codebase, maybe @fpersico can help).

Reply:
Yup, you're right. But right now if all went wrong returning an empty array or an error brings the same result where this function is used: an API call with the missing custom headers and we expect a server error.